modules/gdscript/gdscript_tokenizer.cpp

```diff
@@ -937,8 +937,10 @@ void GDScriptTokenizerText::_advance() {
 								return;
 							}
 							hexa_found = true;
-						} else if (GETCHAR(i) == 'b') {
-							if (hexa_found || bin_found || str.length() != 1 || !((i == 1 && str[0] == '0') || (i == 2 && str[1] == '0' && str[0] == '-'))) {
+						} else if (hexa_found && _is_hex(GETCHAR(i))) {
```

Comment:
Why was this moved though?

Reply:
One of things that caused this bug is the order of the checks. And if it's already in a hex literal and it's a hex character, better to move on sooner than to spend time on checks later.